### PR TITLE
Feature/conversation model

### PR DESCRIPTION
**Pull request by martinjaimem**
Related to the US:

> As a user I should be able to see all my compatible targets so I can start a conversation with any of my compatible users

Done:
- [X] model for conversations an messages
- [X] Specs for models

### CODE REVIEW

File: spec/models/conversation_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      is_expected.to be_valid
+    end
+  end
+
+  context 'validate users qty' do
+    subject { conversation }
+
+    it 'should not save an instance with #users gt 2' do
+      subject.users << user
+      is_expected.not_to be_valid
```
Comment (GuilleLeopold) [inline spec/models/conversation_spec.rb]: you can check on `conversation.errors` and check if the error match

File: app/models/conversation.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  has_many :users, through: :user_conversations
+
+  validate :users_validations
+
+  private
+
+  def users_validations
+    if users.length > 2
+      errors.add(:users, "of conversation must be less than 2, but there were #{users.length}")
+    elsif (users.length == 2) && users[0].id == users[1].id
```
Comment (diebarral) [inline app/models/conversation.rb]: Remove parenthesis

File: app/models/conversation.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  has_many :user_conversations, dependent: :destroy
+  has_many :users, through: :user_conversations
+
+  validate :users_validations
+
+  private
+
+  def users_validations
+    if users.length > 2
+      errors.add(:users, "must be less than 2, but there were #{users.length}")
```
Comment (GuilleLeopold) [inline app/models/conversation.rb]: Sorry I forgot to comment it before, this kind of messages tha are displayed to the user should be added to Internationalization using i18n, check the en.yml file

Comment (martinjaimem) [reply app/models/conversation.rb]: No problem! 🚀

Comment (martinjaimem) [reply app/models/conversation.rb]: I will make another PR to add the I18n configuration to the project

**Pull request merged**